crapssim/bet.py

```diff
@@ -466,87 +472,96 @@ class Hardway(Bet):
             remove = True
-            win_amount = self.payoutratio * self.bet_amount
-        elif dice_object.total in [self.number, 7]:
+            win_amount = self.payout_ratio * self.bet_amount
+        elif self.table.dice.total in [self.number, 7]:
             status = "lose"
             remove = True
 
         return status, win_amount, remove
 
 
 class Hard4(Hardway):
+    payout_ratio: int = 7
+
     def __init__(self, bet_amount: float):
         super().__init__(bet_amount)
         self.name: str = "Hard4"
         self.winning_result: list[int | None] = [2, 2]
         self.number: int = 4
-        self.payoutratio: int = 7
 
 
 class Hard6(Hardway):
+    payout_ratio: int = 9
+
     def __init__(self, bet_amount: float):
         super().__init__(bet_amount)
         self.name: str = "Hard6"
         self.winning_result: list[int | None] = [3, 3]
         self.number: int = 6
-        self.payoutratio: int = 9
 
 
 class Hard8(Hardway):
+    payout_ratio: int = 9
+
     def __init__(self, bet_amount: float):
         super().__init__(bet_amount)
         self.name: str = "Hard8"
         self.winning_result: list[int | None] = [4, 4]
         self.number: int = 8
-        self.payoutratio: int = 9
 
 
 class Hard10(Hardway):
+    payout_ratio: int = 7
+
     def __init__(self, bet_amount: float):
         super().__init__(bet_amount)
         self.name: str = "Hard10"
         self.winning_result: list[int | None] = [5, 5]
         self.number: int = 10
-        self.payoutratio: int = 7
 
 
 class Fire(Bet):
     def __init__(self, bet_amount: float):
-        super().__init__(bet_amount)
+        super().__init__(bet_amount, None)
         self.name: str = "Fire"
         self.bet_amount: float = bet_amount
         self.points_made: list[int] = []
         self.current_point: int | None = None
 
-    def _update_bet(self, table_object: "Table", dice_object: Dice) -> tuple[str | None, float, bool]:
+    def _update_bet(self) -> tuple[str | None, float, bool]:
         status, win_amount, remove = None, 0.0, False
-        if self.current_point is None and dice_object.total in (4, 5, 6, 8, 9, 10):
-            self.current_point = dice_object.total
-        elif self.current_point is not None and self.current_point == dice_object.total:
+        if self.current_point is None and self.table.dice.total in (4, 5, 6, 8, 9, 10):
+            self.current_point = self.table.dice.total
+        elif self.current_point is not None and self.current_point == self.table.dice.total:
             if self.current_point not in self.points_made:
-                self.points_made = list(set(self.points_made + [dice_object.total]))
+                self.points_made = list(set(self.points_made + [self.table.dice.total]))
                 if len(self.points_made) == 4:
-                    status, win_amount, remove = 'win', 24 * self.bet_amount, False
+                    status, win_amount, remove = 'win', self.payout_ratio, False

```

Comment:
Can this portion be re-worked so that `table.payouts['fire_points']` can have 3 points?  One payout I've heard of is `'fire_points': {3: 7, 4: 24, 5: 249, 6: 299}`

Maybe something like this pseudo-code?

```python
if len(self.points_made) in intersection(self.table.payouts['fire_points'], [3,4,5]):
    status, win_amount, remove = 'win', 'self.payout_ratio, False
```